### ISSUE TITLE
Can't upload a photo with Firefox Android

### ISSUE THREAD

**Issue by rmgr** [mastodon/mastodon#22896]
### Steps to reproduce the problem

1. Log in to Mastodon with Firefox Android
2. New post
3. Tap paperclip

### Expected behaviour

Some form of file manager should come up

### Actual behaviour

Nothing

### Detailed description

I've verified it works in Chrome Android. 

### Specifications

Mastodon v4.0.2
Firefox Android 108.1.0 (Build #2015920443), 49ccb63dd+
Oppo Find X2 Lite - ColorOS 11.1

**Comment by deanveloper** [comment]
Unable to reproduce on a Pixel 7 Pro (Android 13), Firefox v108.1.0 (Google Play Store) - does your Firefox app have file access permissions in Android settings?

**Comment by rmgr** [comment]
I double checked the permissions and I do have file storage allowed. Interestingly I tried disabling file storage permissions and then when I hit the paperclip it prompted me to allow permissions but when I did, nothing happens. 

The file upload has worked on the most up-to-date v3 Mastodon on this phone/browser. 

I've checked with another site (itch.io developer dashboard) and the file upload functions as expected.
